scripts/setup_osx.sh

```diff
@@ -1,38 +1,267 @@
+        exit 0
+        ;;
+      *)
+        _PRINT_HELP=yes die "FATAL ERROR: Got an unexpected argument '$1'" 1
+        ;;
+    esac
+    shift
+  done
+}
 
-echo "Setting up MuJoCo..."
-/bin/bash ./scripts/setup_mujoco.sh
-if [[ "${?}" -ne 0 ]]; then
-  echo -e "\e[0;31mError: MuJoCo couldn't be set up\e[0m"
-  exit 1
+
+parse_commandline "$@"
+### END OF CODE GENERATED BY Argbash (sortof) ### ])
+
+# Utility functions
+script_dir_path() {
+  SCRIPT_DIR="$(dirname ${0})"
+  [[ "${SCRIPT_DIR}" = /* ]] && echo "${SCRIPT_DIR}" || \
+    echo "${PWD}/${SCRIPT_DIR#./}"
+}
+
+# red text
+print_error() {
+  echo -e "\033[0;31m${@}\033[0m"
+}
+
+# yellow text
+print_warning() {
+  echo -e "\033[0;33m${@}\033[0m"
+}
+
+# Obtain the OS X version
+VER="$(sw_vers -productVersion)"
+
+if [[ ! " ${VERIFIED_OSX_VERSIONS[@]} " =~ " ${VER} " ]]; then
+  print_warning "You are attempting to install garage on a version of OS X" \
+    "which we have not verified is working." | fold -s
+  print_warning "\ngarage relies on community contributions to support OS X\n"
+  print_warning "If this installation is successful, please add your OS X" \
+    "version to VERIFIED_OSX_VERSIONS to" \
+    "https://github.com/rlworkgroup/garage/blob/master/scripts/setup_osx.sh" \
+    "on GitHub and submit a pull request to rlworkgroup/garage to help out" \
+    "future users. If the installation is not initially successful, but you" \
+    "find changes which fix it, please help us out by submitting a PR with" \
+    "your updates to the setup script." \
+    | fold -s
+  while [[ "${continue_var}" != "y" ]]; do
+    read -p "Continue? (y/n): " continue_var
+    if [[ "${continue_var}" = "n" ]]; then
+      exit
+    fi
+  done
 fi
 
-echo "Creating conda environment..."
+# Verify there's a file in the mjkey path
+test -f "${_arg_mjkey}" || _PRINT_HELP=yes die \
+  "The path ${_arg_mjkey} of the MuJoCo key is not valid." 1
+
+# Make sure that we're under the garage directory
+GARAGE_DIR="$(dirname $(script_dir_path))"
+cd "${GARAGE_DIR}"
+
+# File where environment variables are stored
+BASH_PROF="${HOME}/.bash_profile"
+
+# Install dependencies
+echo "Installing garage dependencies"
+echo "You will probably be asked for your sudo password"
+
+# Homebrew is required first to install the other dependencies
+hash brew 2>/dev/null || {
+  # Install the Xcode Command Line Tools
+  xcode-select --install
+  # Install Homebrew
+  /usr/bin/ruby -e "$(curl -fsSL \
+    https://raw.githubusercontent.com/Homebrew/install/master/install)"
+}
+
+# For installing garage: bzip2, git, glfw, unzip, wget
+# For building glfw: cmake
+# Required for OpenAI gym: cmake boost boost-python ffmpeg sdl2 swig wget
+# Required for OpenAI baselines: cmake openmpi
+brew update
+brew install \
+  bzip2 \
+  git \
+  glfw \
+  unzip \
+  wget \
+  cmake \
+  boost \
+  boost-python \
+  ffmpeg \
+  sdl2 \
+  swig \
+  openmpi
+# Required by MuJoCo
+brew reinstall gcc@7 --without-multilib
```

Comment:
why reinstall? and why not just install it before all the brew dependencies (which might also need to be compiled using the older GCC).

Also, unless something needs a higher version, gcc6 is more broadly supported than gcc7, which is why I picked it.

mine worked with `brew install gcc@6 --without-multilib` before all the other brew deps